### DISCUSSION TITLE
identical useMutation instances sharing status

### DISCUSSION THREAD

**Discussion by babbins** [TanStack/query#818] [General]
Is there a way for two identical instances of useMutation could share a single `status` so I can share the status across two different parts of my app without having to hoist my useMutation to parent and pass via props? This seems like it would only be possible with some sort of key identifier being passed into useMutation like useQuery's cache key. Here's a sandbox illustrating my problem:

https://codesandbox.io/s/react-query-rtjjo?file=/src/index.js

**Comment by wolverineks** [comment]
you could raises it up into a context with the same interface

**Reply by wolverineks** [reply]
```ts
...
<MutationContext.Provider value={useMutation(mutationFn)}>{children}</MutationContext.Provider>
...

const useMutation = () => React.useContext(MutationContext)
```
